File: 0049-group-anagrams/0049-group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        anagram_dict = {}
        
        for string in strs:
            sorted_str = "".join(sorted(string))
            
            if sorted_str not in anagram_dict:
                anagram_dict[sorted_str] = []
            
            anagram_dict[sorted_str].append(string)
            
        return list(anagram_dict.values())

File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    anagrams = {}
    
    for (let ele of strs) {
        sorted = (ele.split('')).sort()
        if (!anagrams[sorted]) {
            anagrams[sorted] = []
        }
        anagrams[sorted].push(ele)
    }
    
    return Object.values(anagrams)
};


// create hashtable, key:pair => "aet":[]
// iterate thru strs, sort the ele and add to hashtable
// return values of hashtable